test/Dialect/Comb/basic.mlir

```diff
@@ -0,0 +1,11 @@
+// RUN: circt-opt %s | FileCheck %s
+
+hw.type_scope @__hw_typedecls {
+  hw.typedecl @foo : i1
+}
+
+// https://github.com/llvm/circt/issues/5772
+// CHECK-LABEL: @Issue5772
+hw.module @Issue5772(in %arg0: !hw.typealias<@__hw_typedecls::@foo,i1>) {
+  %0 = comb.concat %arg0 : !hw.typealias<@__hw_typedecls::@foo,i1>
```

Comment:
Since passing this example through circt-opt is enough to trigger the crash, in a test like this we are mainly checking that circt-opt is able to parse the IR and then emit it unchanged. So in that case, I think it would be reasonable to add a CHECK line that this concat op is present in the output unchanged, which could be as simple as:

```
// CHECK: comb.concat %arg0 : !hw.typealias<@__hw_typedecls::@foo,i1>
```

Reply:
Thanks for the rationale, I pushed your proposal!